app/code/Magento/Wishlist/Test/Unit/Controller/Index/UpdateTest.php

```diff
@@ -0,0 +1,252 @@
+use Magento\Wishlist\Controller\Index\Update;
+use Magento\Wishlist\Controller\WishlistProviderInterface;
+use Magento\Wishlist\Helper\Data;
+use Magento\Wishlist\Model\Item;
+use Magento\Wishlist\Model\LocaleQuantityProcessor;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test for upate controller wishlist
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class UpdateTest extends TestCase
+{
+    /**
+     * @var Validator $formKeyValidator
+     */
+    private $formKeyValidator;
+
+    /**
+     * @var WishlistProviderInterface $wishlistProvider
+     */
+    private $wishlistProvider;
+
+    /**
+     * @var LocaleQuantityProcessor $quantityProcessor
+     */
+    private $quantityProcessor;
+
+    /**
+     * @var Update $updateController
+     */
+    private $updateController;
+
+    /**
+     * @var $context
+     */
+    private $context;
+
+    /**
+     * @var Redirect $resultRedirect
+     */
+    private $resultRedirect;
+
+    /**
+     * @var ResultFactory $resultFatory
+     */
+    private $resultFactory;
+
+    /**
+     * @var RequestInterface $requestMock
+     */
+    private $requestMock;
+
+    /**
+     * @var ObjectManagerInterface $objectManagerMock
+     */
+    private $objectManagerMock;
+
+    /**
+     * @var ManagerInterface $messageManager
+     */
+    private $messageManager;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $this->formKeyValidator = $this->createMock(Validator::class);
+        $this->wishlistProvider = $this->createMock(WishlistProviderInterface::class);
+        $this->quantityProcessor = $this->createMock(LocaleQuantityProcessor::class);
+        $this->context = $this->createMock(Context::class);
+        $this->resultRedirect = $this->createMock(Redirect::class);
+        $this->resultFactory = $this->createPartialMock(ResultFactory::class, ['create']);
+        $this->requestMock = $this->getMockBuilder(RequestInterface::class)
+            ->setMethods(['getPostValue'])
+            ->getMockForAbstractClass();
+        $this->objectManagerMock = $this->createMock(ObjectManagerInterface::class);
+
+        $this->context->expects($this->once())
+            ->method('getResultFactory')
+            ->willReturn($this->resultFactory);
+        $this->context->expects($this->once())
+            ->method('getObjectManager')
+            ->willReturn($this->objectManagerMock);
+
+        $this->resultFactory->expects($this->any())
+                              ->method('create')
+                              ->willReturn($this->resultRedirect);
+        $this->context->expects($this->any())
+            ->method('getRequest')
+            ->willReturn($this->requestMock);
+
+        $this->messageManager = $this->createMock(ManagerInterface::class);
+        $this->context->expects($this->any())
+            ->method('getMessageManager')
+            ->willReturn($this->messageManager);
+
+        $this->updateController = new Update(
```

Comment:
Use ObjectManager helper for building the tested object.

Reply:
ok